### PR TITLE
azureml-pipeline-databricks

### PR DESCRIPTION
**Pull request by algattik**
Automating the provisioning and execution of an Azure ML Pipeline that executes a Databricks notebook doing data preparation.

This sample automates the provisioning of an ML execution environment using Terraform, and the provisioning and execution of an Azure ML Pipeline that runs a Databricks notebook doing data engineering.

This sample demonstrates:
* Deployment of Azure ML and Databricks infrastructure using Terraform (based on the [Terraform Azure DevOps starter sample](https://github.com/microsoft/terraform-azure-devops-starter)).
* Provisioning of Databricks accounts and notebooks with Azure AD authentication, using the [databricks-client](https://pypi.org/project/databricks-client/) module.
* Unit testing of Databricks notebooks with PySpark, using the [databricks-test](https://pypi.org/project/databricks-test/) module.
* Integrating a [Databricks step](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-create-your-first-pipeline#databricks) into an Azure ML pipeline.

Meant to be combined with https://github.com/microsoft/MLOpsPython in order to add ETL / feature engineering to an ML training pipeline.

### CODE REVIEW

File: Python/azureml-pipeline-databricks/README.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+We use Azure DevOps for running our MLOps pipeline with build (CI), ML training and scoring service release
+(CD) stages. If you don't already have an Azure DevOps organization, create one by
+following the instructions [here](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/create-organization?view=azure-devops).
+
+If you already have an Azure DevOps organization, create a [new project](https://docs.microsoft.com/en-us/azure/devops/organizations/projects/create-project?view=azure-devops).
+
+### Create an ARM service connection for Terraform
+
+The `DataOpsML ARM Connection` service connection is used by the [Azure DevOps pipeline](environment_setup/terraform-init-template.yml) to creates the Azure ML workspace and associated resources through Terraform. The pipeline requires an **Azure Resource Manager**
```
Comment (Anaig) [inline Python/azureml-pipeline-databricks/README.md]:
```suggestion
The `DataOpsML ARM Connection` service connection is used by the [Azure DevOps pipeline](environment_setup/terraform-init-template.yml) to create the Azure ML workspace and associated resources through Terraform. The pipeline requires an **Azure Resource Manager**
```

File: Python/azureml-pipeline-databricks/README.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+[service connection](https://docs.microsoft.com/en-us/azure/devops/pipelines/library/service-endpoints?view=azure-devops&tabs=yaml#create-a-service-connection).
```
Comment (Anaig) [inline Python/azureml-pipeline-databricks/README.md]:
```suggestion
[service connection](https://docs.microsoft.com/en-us/azure/devops/pipelines/library/service-endpoints?view=azure-devops&tabs=yaml#create-a-service-connection) at the subscription level.
```

File: Python/azureml-pipeline-databricks/README.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+**Note:** Creating the ARM service connection scope requires 'Owner' or 'User Access Administrator' permissions on the subscription.
+You must also have sufficient permissions to register an application with
+your Azure AD tenant, or receive the ID and secret of a service principal
+from your Azure AD Administrator. That principal must have 'Contributor'
+permissions on the subscription.
+
+### Create a storage account for the Terraform state
+
+Create an Azure storage account. In the storage account, create a storage container named `terraformstate`.
```
Comment (Anaig) [inline Python/azureml-pipeline-databricks/README.md]:
```suggestion
Create an Azure storage account. using the Portal interface or the CLI command below In the storage account, create a storage container named `terraformstate`.
```
```
location=[LOCATION]
rg=[RG_NAME]
storage_name=[UNIQUE_STORAGE_NAME]
az group create --location $location --name $rg
az storage account create --name $storage_name --resource-group $rg --location $location
az storage container create -name terraformstate --account-name $storage_name
```

File: Python/azureml-pipeline-databricks/README.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+the BASE_NAME value should not exceed 10 characters and it should contain numbers and lowercase letters only.
+
+Make sure to select the **Allow access to all pipelines** checkbox in the
+variable group configuration.
+
+### Run the Terraform pipeline
+
+In your [Azure DevOps](https://dev.azure.com) project create a new build
+pipeline referring to the
+[environment_setup/terraform-pipeline.yml](../environment_setup/terraform-pipeline.yml)
```
Comment (Anaig) [inline Python/azureml-pipeline-databricks/README.md]:
```suggestion
[environment_setup/terraform-pipeline.yml](./environment_setup/terraform-pipeline.yml)
```

File: Python/azureml-pipeline-databricks/README.md (hunk 3 above)
Comment (algattik) [reply Python/azureml-pipeline-databricks/README.md]: For standardization, linking to doc page rather than adding doc

**Pull request merged**